server/src/main/java/org/apache/druid/segment/indexing/granularity/GranularitySpec.java

```diff
@@ -41,11 +40,11 @@
 public interface GranularitySpec
 {
   /**
-   * Set of all time groups, broken up on segment boundaries. Should be sorted by interval start and non-overlapping.
+   * Iterable all time groups, broken up on segment boundaries. Should be sorted by interval start and non-overlapping.
    *
-   * @return set of all time groups
+   * @return Iterable of all time groups
    */
-  Optional<SortedSet<Interval>> bucketIntervals();
+  Iterable<Interval> bucketIntervals();
```

Comment:
Suggest `sortedBucketIntervals()` to make it clear the results are sorted.

Reply:
Done